react/week4/my-app/src/Inputuser.js

```diff
@@ -0,0 +1,81 @@
+import React, { useState, useEffect, useCallback } from "react";
+
+import ShowUser from "./showusers";
+
+const InputUser = () => {
+  const [name, setName] = useState("");
+
+  const [users, setUsers] = useState([]);
+
+  const [isLoading, setIsLoading] = useState(true);
+  
+  const [isEmpty, setEmpty] = useState(true);
+
+  const apiBaseUrl = "https://api.github.com/search/users?q=";
+
+  const SearchUser = useCallback((value) => {
+    setUsers([]);
+    setEmpty(true);
+
+    let apiUrl = apiBaseUrl + value;
+    console.log(apiUrl);
+    fetch(apiUrl)
+      .then((res) => res.json())
+      .then((result) => {
+        console.log(result);
+
+        console.log(result.items);
+
+        if (result.items) {
+          const items = result.items
+            .map((item) => item.login)
+            .filter((item) => item.startsWith(value));
+
+          setUsers((prev) => {
+            setIsLoading(false);
+            setEmpty(false);
+
+            return prev.concat(items);
+          });
+
+          console.log(users);
+        }
+      });
+    console.log(users);
+  }, []);
+
+  useEffect(() => {
+    SearchUser();
+  }, [SearchUser]);
```

Comment:
This can be removed as you are called the SearchUser() function in the onChange handler